aerospike-mixin/dashboards/aerospike-overview.libsonnet

```diff
@@ -0,0 +1,1176 @@
+      datasource=promDatasource,
+      legendFormat='{{aerospike_cluster}}',
+      format='time_series',
+    ),
+  ],
+  type: 'bargauge',
+  title: 'Unavailable partitions',
+  description: 'Number of unavailable data partitions in an Aerospike cluster.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'thresholds',
+      },
+      decimals: 0,
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+        ],
+      },
+      unit: 'none',
+    },
+    overrides: [],
+  },
+  options: {
+    displayMode: 'gradient',
+    minVizHeight: 10,
+    minVizWidth: 0,
+    orientation: 'auto',
+    reduceOptions: {
+      calcs: [
+        'lastNotNull',
+      ],
+      fields: '',
+      values: false,
+    },
+    showUnfilled: true,
+    text: {
+      valueSize: 100,
+    },
+    valueMode: 'color',
+  },
+  pluginVersion: '10.2.0-59542pre',
+  transformations: [
+    {
+      id: 'sortBy',
+      options: {
+        fields: {},
+        sort: [
+          {
+            desc: true,
+            field: 'Time',
+          },
+        ],
+      },
+    },
+  ],
+};
+
+local deadPartitionsPanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'sum by(job, aerospike_cluster) (aerospike_namespace_dead_partitions{job=~"$job", aerospike_cluster=~"$aerospike_cluster"})',
+      datasource=promDatasource,
+      legendFormat='{{aerospike_cluster}}',
+    ),
+  ],
+  type: 'bargauge',
+  title: 'Dead partitions',
+  description: 'Number of dead data partitions in an Aerospike cluster.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'thresholds',
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+        ],
+      },
+      unit: 'none',
+    },
+    overrides: [],
+  },
+  options: {
+    displayMode: 'gradient',
+    minVizHeight: 10,
+    minVizWidth: 0,
+    orientation: 'auto',
+    reduceOptions: {
+      calcs: [
+        'lastNotNull',
+      ],
+      fields: '',
+      values: false,
+    },
+    showUnfilled: true,
+    text: {
+      valueSize: 100,
+    },
+    valueMode: 'color',
+  },
+  pluginVersion: '10.2.0-59542pre',
+  transformations: [
+    {
+      id: 'sortBy',
+      options: {
+        fields: {},
+        sort: [
+          {
+            desc: true,
+            field: 'Time',
+          },
+        ],
+      },
+    },
+  ],
+};
+
+local topNodesByMemoryUsagePanel = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'topk($k, 100 - sum by(aerospike_cluster, instance) (avg_over_time(aerospike_node_stats_system_free_mem_pct{job=~"$job", aerospike_cluster=~"$aerospike_cluster"}[$__interval])))',
```

Comment:
i'm probably a little behind on this. Is this supposed to be showing the top % of "worst nodes"? or all nodes that have a metric value > some %?

either way, can you help explain how to me how this calculation does either of those things, because my brain isn't processing it right now?

Reply:
It takes the free pct for each node, subtracts 100 to get the usage, then takes the top-k usages.

Reply:
Just to clarify, Gabriel directed me to use top-k here instead of a percentile like he suggested earlier.